### PR TITLE
feat: KalixClient.post without body

### PR DESCRIPTION
**Pull request by octonato**
References https://github.com/lightbend/kalix/issues/7824

### CODE REVIEW

**Comment by octonato** [comment]
@efgpinto, I picked this one because it's a low-hanging fruit, but then realised that you made the point that we should probably hold-on with it before we decide what to do with https://github.com/lightbend/kalix-jvm-sdk/issues/1173 

We can wait for merging it. Anyway, we can discuss it tomorrow.

**Comment by efgpinto** [comment]
> @efgpinto, I picked this one because it's a low-hanging fruit, but then realised that you made the point that we should probably hold-on with it before we decide what to do with #1173

Yeah, I add a similar thought today... That it would be good to include this for the release. I think we can just have a quick brainstorm about it tomorrow but I still think this might be the way forward.

**Review by efgpinto — COMMENTED** [review]
I think we should do the same for `PUT`.

Also, if you search for `"",` you can probably find 2 or 3 places where you can apply the new version. Cleaner and validation as bonus.

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala [hunk 1]
```diff
@@ -151,12 +229,10 @@ trait KalixClient {
...
    *   The type returned by the target endpoint
-   * @tparam P
-   *   Type used as a body for the request
    * @tparam R
    *   Type returned as response from the target endpoint
    * @return
    *   a [[kalix.javasdk.DeferredCall]] to be used in forwards and timers or to be executed in place
    */
-  def delete[P, R](uri: String, body: P, returnType: Class[R]): DeferredCall[Any, R]
+  def delete[R](uri: String, returnType: Class[R]): DeferredCall[Any, R]
```
Comment (octonato) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala:237]: @efgpinto, I removed the body for delete just to follow the WebClient choices. 

We can add it later if eventually, but happy to bring it back it there is a strong reason for it.

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala [hunk 2]
```diff
@@ -76,6 +76,32 @@ trait KalixClient {
...
+   *       var serviceCall = kalixClient.post("/counter/create", Confirmation.class);
+   *       return effects().forward(serviceCall);
+   *     }
+   * }}}
+   *
+   * @param uri
+   *   The resource path where the target endpoint will be reached at. Query parameters can be passed in as part of the
+   *   URI but should be encoded if containing special characters.
+   * @param body
+   *   The HTTP body type expected by the target endpoint
```
Comment (efgpinto) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala:95]:
```suggestion
```

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala [hunk 3]
```diff
@@ -104,6 +130,32 @@ trait KalixClient {
...
+   *       var serviceCall = kalixClient.put("/fibonacci/next", number, Number.class);
+   *       return effects().forward(serviceCall);
+   *     }
+   * }}}
+   *
+   * @param uri
+   *   The resource path where the target endpoint will be reached at. Query parameters can be passed in as part of the
+   *   URI but should be encoded if containing special characters.
+   * @param body
+   *   The HTTP body type expected by the target endpoint
```
Comment (efgpinto) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala:149]:
```suggestion
```

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala [hunk 4]
```diff
@@ -132,14 +184,40 @@ trait KalixClient {
...
+   *       var serviceCall = kalixClient.patch("/user/" + user.id + "/email", email, User.class);
+   *       return effects().forward(serviceCall);
+   *     }
+   * }}}
+   *
+   * @param uri
+   *   The resource path where the target endpoint will be reached at. Query parameters can be passed in as part of the
+   *   URI but should be encoded if containing special characters.
+   * @param body
+   *   The HTTP body type expected by the target endpoint
```
Comment (efgpinto) [inline sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala:203]:
```suggestion
```

**Review by efgpinto — APPROVED** [review]
LGTM (with suggestion of removing leftovers in javadoc)

File: sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala (hunk 2 above)
Comment (octonato) [reply sdk/spring-sdk/src/main/scala/kalix/springsdk/KalixClient.scala:95]: Of course, I was only paying attention to the P param. 🤦

**Pull request merged**